File: IntegrationCore/Models/DB/ConnectionFieldDefinition.cs
using System.Collections.Generic;

namespace IntegrationCore.Models.DB
{
    public class ConnectionFieldDefinition
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public FieldTransferRole Role { get; set; }
        public int SystemId { get; set; }
        public SystemDefinition SystemDefinition { get; set; }
        public IEnumerable<ConnectionFieldValue> ConnectionFieldValues { get; set; }
    }

    public enum FieldTransferRole
    {
        Header,
        JsonBody,
        Query
    }
}

File: IntegrationCore/ClientApp/src/app/models/connection-field-definition.ts
import { SystemDefinition } from './system-definition';
import { ConnectionFieldValue } from './connection-field-value';

export class ConnectionFieldDefinition {
  id?: number;
  name: string;
  role: FieldTransferRole;
  systemId?: number;
  systemDefinition?: SystemDefinition;
  connectionFieldValues?: ConnectionFieldValue[];
}

export enum FieldTransferRole {
  Header,
  JsonBody,
  Query
}
